src/Providers/Http/DTO/Response.php

```diff
@@ -0,0 +1,267 @@
+<?php
+
+declare(strict_types=1);
+
+namespace WordPress\AiClient\Providers\Http\DTO;
+
+use InvalidArgumentException;
+use WordPress\AiClient\Common\AbstractDataTransferObject;
+
+/**
+ * Represents an HTTP response.
+ *
+ * This class encapsulates HTTP response data that has been converted
+ * from PSR-7 responses by the HTTP transporter.
+ *
+ * @since n.e.x.t
+ *
+ * @phpstan-type ResponseArrayShape array{
+ *     statusCode: int,
+ *     headers: array<string, string|list<string>>,
+ *     body?: string|null,
+ *     reasonPhrase: string
+ * }
+ *
+ * @extends AbstractDataTransferObject<ResponseArrayShape>
+ */
+class Response extends AbstractDataTransferObject
+{
+    public const KEY_STATUS_CODE = 'statusCode';
+    public const KEY_HEADERS = 'headers';
+    public const KEY_BODY = 'body';
+    public const KEY_REASON_PHRASE = 'reasonPhrase';
+
+    /**
+     * @var int The HTTP status code.
+     */
+    protected int $statusCode;
+
+    /**
+     * @var array<string, string|list<string>> The response headers.
+     */
+    protected array $headers;
+
+    /**
+     * @var string|null The response body.
+     */
+    protected ?string $body;
+
+    /**
+     * @var string The reason phrase.
+     */
+    protected string $reasonPhrase;
+
+    /**
+     * Constructor.
+     *
+     * @since n.e.x.t
+     *
+     * @param int $statusCode The HTTP status code.
+     * @param array<string, string|list<string>> $headers The response headers.
```

Comment:
This is something I had mentioned somewhere in #46, we should go with one consistent format for headers. I think `array<string, list<string>>` makes sense.